### ISSUE TITLE
GPU requirement

### ISSUE THREAD

**Issue by S-Watanabe0123** [KumarLabJax/deep-hrnet-mouse#4]
My laptop has Nvidia GeForce RTX 3060 (6GB), yet this was not enough to run infermousepose.py. I used another computer with RTX 2070 (8GB), but this was not enough either. Is there any way to reallocate the memory to run this? or does it require a high-end GPU?

![Screenshot from 2023-05-26 15-47-04](https://github.com/KumarLabJax/deep-hrnet-mouse/assets/90536715/1b5ddf80-1df7-4ddd-8122-e84fbd50adfa)

**Comment by SkepticRaven** [comment]
The way this network is implemented, it will scale in memory based on the requested input size.

What frame size of the video you are trying? Our 480x480 videos definitely fit on a 8GB VRAM GPU. It's been awhile since we ran the tests, but I think we were able to run it on 4GB GPUs. 2GB was too small.

We use `ffmpeg` to do most of our video manipulation tasks. A potential command for this could be (assuming your video is square - would need to be changed to preserve the aspect ratio):
```
ffmpeg -i input_video.avi -vf scale=480:480 -q 0 scaled_input.avi
```

Last thing worth noting is that the network tends to work the best when mice are about 50px long when walking due to how the network was trained. There is some tolerance to size, but we haven't fully characterized it. So while scaling down the input will both make the network run faster and use a smaller memory footprint, it may also change performance.
